client/landing/stepper/declarative-flow/internals/hooks/use-launchpad-decider/index.tsx

```diff
@@ -0,0 +1,114 @@
+import { updateLaunchpadSettings } from '@automattic/data-stores';
+import { ExperimentAssignment } from '@automattic/explat-client';
+import { useExperiment } from 'calypso/lib/explat';
+
+export const LAUNCHPAD_EXPERIMENT_NAME = 'calypso_onboarding_launchpad_removal_test_2024_08';
+
+interface Props {
+	exitFlow: ( path: string ) => void;
+	navigate: ( path: string ) => void;
+}
+
+interface PostFlowUrlProps {
+	flow: string;
+	siteId: string | number | null;
+	siteSlug: string;
+}
+
+interface SiteProps {
+	siteId: string | number | null;
+	siteSlug: string | null;
+}
+
+export const useLaunchpadDecider = ( { exitFlow, navigate }: Props ) => {
+	const [ isLoadingExperiment, experimentAssignment ] = useExperiment( LAUNCHPAD_EXPERIMENT_NAME );
+	const showCustomerHome = shouldShowCustomerHome( isLoadingExperiment, experimentAssignment );
+
+	let launchpadStateOnSkip: null | 'skipped' = null;
+	if ( showCustomerHome ) {
+		launchpadStateOnSkip = 'skipped';
+	}
+
+	const setLaunchpadSkipState = ( siteIdOrSlug: string | number | null ) => {
+		if ( siteIdOrSlug && launchpadStateOnSkip ) {
+			updateLaunchpadSettings( siteIdOrSlug, { launchpad_screen: launchpadStateOnSkip } );
+		}
+	};
+
+	return {
+		getPostFlowUrl: ( { flow, siteId, siteSlug }: PostFlowUrlProps ) => {
+			if ( showCustomerHome ) {
+				return '/home/' + siteSlug;
+			}
+
+			if ( siteId ) {
+				return `/setup/${ flow }/launchpad?siteSlug=${ siteSlug }&siteId=${ siteId }`;
+			}
+
+			return `/setup/${ flow }/launchpad?siteSlug=${ siteSlug }`;
+		},
+		postFlowNavigator: ( { siteId, siteSlug }: SiteProps ) => {
+			if ( showCustomerHome ) {
+				setLaunchpadSkipState( siteId || siteSlug );
+
+				exitFlow( '/home/' + siteSlug );
+				return;
+			}
+
+			navigate( `launchpad?siteSlug=${ siteSlug }&siteId=${ siteId }` );
+			return;
+		},
+		initializeLaunchpadState: ( { siteId, siteSlug }: SiteProps ) => {
+			if ( showCustomerHome ) {
+				setLaunchpadSkipState( siteId || siteSlug );
+				return;
+			}
+		},
+	};
+};
+
+/**
+ * Determine if the customer home should be shown
+ * @param isLoadingExperiment
+ * @param experimentAssignment
+ */
+export function shouldShowCustomerHome(
+	isLoadingExperiment: boolean,
+	experimentAssignment: ExperimentAssignment | null
+): boolean {
+	return (
+		! isLoadingExperiment &&
+		( 'treatment' === experimentAssignment?.variationName ||
+			// for testing/development purposes we can use sessionStorage to force the treatment
+			'treatment' ===
+				window.sessionStorage.getItem( 'launchpad_removal_2024_experiment_variation' ) )
```

Comment:
will this eventually get removed from the code changes I see we can force the experiment variation through client side ?

Reply:
The session item is only for our dev/testing purposes. We are setting it manually via the developer tool. I didn't find a better way to test it.
Once we start the experiment, I can remove it in an additional PR.

Reply:
Since the experiment is running, I created PR for removing dev/testing piece of code (we don't need it anymore): #94764

Reply:
Thanks for the update!